dali/core/mm/default_resource_test.cc

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include "dali/core/mm/default_resources.h"
+#include "dali/core/mm/detail/align.h"
+#include "dali/core/cuda_stream.h"
+#include "dali/core/cuda_event.h"
+#include "dali/core/cuda_error.h"
+#include "dali/core/device_guard.h"
+
+namespace dali {
+namespace mm {
+namespace test {
+
+TEST(MMDefaultResource, Host) {
+  const auto &rsrc = GetDefaultResource<memory_kind::host>();
+  ASSERT_NE(rsrc, nullptr);
+  char *mem = static_cast<char*>(rsrc->allocate(1000, 32));
+  ASSERT_NE(mem, nullptr);
+  EXPECT_TRUE(mm::detail::is_aligned(mem, 32));
+  memset(mem, 42, 1000);
+  rsrc->deallocate(mem, 1000, 32);
```

Comment:
Do you need that call?

Reply:
I do, it's an actual allocation from a real resource.

Reply:
I'm asking if the destructor won't do that under the hood when you leave the test scope.

Reply:
No, there's no destructor involved. I'm getting a pointer to current global resource and allocating some memory - I need to free that memory, otherwise the test would create a memory leak.

Reply:
ok